config/make.go

```diff
@@ -0,0 +1,136 @@
+package config
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"github.com/Shopify/sarama"
+	"github.com/pkg/errors"
+)
+
+func Make() (*Config, error) {
+	envKafkaConfig, err := GetKafka()
+	if err != nil {
+		return nil, err
+	}
+
+	saramaConfig := sarama.NewConfig()
+
+	err = ConfigureSarama(envKafkaConfig, saramaConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	config := &Config{
+		Kafka:         saramaConfig,
+		Brokers:       envKafkaConfig.Brokers,
+		ConsumerGroup: envKafkaConfig.ConsumerGroup,
+		Topic:         envKafkaConfig.Topic,
+	}
+
+	return config, nil
+}
+
+func ConfigureSarama(kafkaConfig EnvKafkaConfig, saramaConfig *sarama.Config) error {
+	err := authentication(saramaConfig, kafkaConfig)
+	if err != nil {
+		return errors.Wrap(err, "Error parsing Kafka authentication: %v")
+	}
+
+	version, err := sarama.ParseKafkaVersion(kafkaConfig.Version)
+	if err != nil {
+		return errors.Wrap(err, "Error parsing Kafka version: %v")
+	}
+
+	saramaConfig.Version = version
+
+	switch kafkaConfig.Assignor {
+	case "sticky":
+		saramaConfig.Consumer.Group.Rebalance.Strategy = sarama.BalanceStrategySticky
+	case "roundrobin":
+		saramaConfig.Consumer.Group.Rebalance.Strategy = sarama.BalanceStrategyRoundRobin
+	case "range":
+		saramaConfig.Consumer.Group.Rebalance.Strategy = sarama.BalanceStrategyRange
+	default:
+		err = errors.New("Unrecognized consumer group partition assignor: %s")
+	}
+
+	if kafkaConfig.OldestFirst {
+		saramaConfig.Consumer.Offsets.Initial = sarama.OffsetOldest
+	}
+
+	return nil
+}
+
+func authentication(saramaConfig *sarama.Config, kafkaConfig EnvKafkaConfig) error {
+	if kafkaConfig.AuthType == "ssl" {
+		err := sslAuthentication(saramaConfig, kafkaConfig)
+		if err != nil {
+			return err
+		}
+
+		return nil
```

Comment:
```suggestion
		return sslAuthentication(saramaConfig, kafkaConfig)
```

Este método `sslAuthentication` retorna `nil`  se não tiver erro?, se sim vc poderia retornar err direto